sdk/examples/xo_python/sawtooth_xo/processor/handler.py

```diff
@@ -38,211 +41,95 @@ def family_versions(self):
-                'Invalid Action: space {} already taken'.format(space))
+        elif xo_payload.action == 'create':
 
-    elif action == 'delete':
-        if board is None:
-            raise InvalidTransaction('Invalid action: game does not exist')
+            if xo_state.get_game(xo_payload.name) is not None:
+                raise InvalidTransaction(
+                    'Invalid action: Game already exists: {}'.format(
+                    xo_payload.name))
 
+            game = Game(name=xo_payload.name,
+                        board="-" * 9,
+                        state="P1-NEXT",
+                        player1="",
+                        player2="")
 
-def _make_xo_address(namespace_prefix, name):
-    return namespace_prefix + \
-        hashlib.sha512(name.encode('utf-8')).hexdigest()[:64]
-
-
-def _get_state_data(context, namespace_prefix, name):
-    # Get data from address
-    state_entries = \
-        context.get_state([_make_xo_address(namespace_prefix, name)])
-
-    # context.get_state() returns a list. If no data has been stored yet
-    # at the given address, it will be empty.
-    if state_entries:
-        try:
-            state_data = state_entries[0].data
-
-            game_list = {
-                name: (board, state, player1, player2)
-                for name, board, state, player1, player2 in [
-                    game.split(',')
-                    for game in state_data.decode().split('|')
-                ]
-            }
-
-            board, state, player1, player2 = game_list[name]
-
-        except ValueError:
-            raise InternalError("Failed to deserialize game data.")
-
-    else:
-        game_list = {}
-        board = state = player1 = player2 = None
-
-    return board, state, player1, player2, game_list
-
-
-def _store_state_data(
-        context, game_list,
-        namespace_prefix, name,
-        board, state, player1, player2):
-
-    game_list[name] = board, state, player1, player2
-
-    state_data = '|'.join(sorted([
-        ','.join([name, board, state, player1, player2])
-        for name, (board, state, player1, player2) in game_list.items()
-    ])).encode()
-
-    addresses = context.set_state(
-        {_make_xo_address(namespace_prefix, name): state_data})
-
-    if len(addresses) < 1:
-        raise InternalError("State Error")
-
-
-def _delete_game(context, name, namespace_prefix):
-    LOGGER.warning('Deleting game %s', name)
-
-    address = _make_xo_address(namespace_prefix, name)
-
-    addresses = context.delete_state([address])
+            xo_state.set_game(game=game)
+            _display("Player {} created a game.".format(signer[:6]))
 
-    if not addresses:
-        raise InternalError('State delete error')
+        elif xo_payload.action == 'take':
+            game = xo_state.get_game(xo_payload.name)
 
+            if game is None:
+                raise InvalidTransaction(
+                    'Invalid action: Take requires an existing game.')
```

Comment:
Let's lose the ending periods since it's not friendly to chaining exception error messages together. (Ideally as another commit since this is a hold-over from the previous code.)